value_test.go

```diff
@@ -97,6 +98,44 @@ func TestValueString(t *testing.T) {
 	}
 }
 
+func TestValueString_GoToJSAndBack(t *testing.T) {
+	t.Parallel()
+	ctx := v8.NewContext(nil)
+	iso := ctx.Isolate()
+	defer iso.Dispose()
+	defer ctx.Close()
+
+	str := "s\x00s\x00"
+	jsStr, err := v8.NewValue(iso, str)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test whether the go->js keeps the null chars
+	val, err := ctx.RunScript("(str) => { return str === String.fromCharCode(115, 0, 115, 0)}", "test.js")
```

Comment:
The [arrow function syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) allows for a less verbose syntax for a single argument and for a single line body

```suggestion
	val, err := ctx.RunScript(`str => str === "s\x00s\x00"}`, "test.js")
```

It looks like NewValue isn't well tested, so this could be turned into a table driven test with `input` and `predicate` fields.  The js->go part was already tested in TestValueString, so this test could be renamed to TestNewValue

Reply:
1942901